src/Authenticator/JwtAuthenticator.php

```diff
@@ -18,15 +18,19 @@
 
 use ArrayObject;
 use Authentication\Identifier\IdentifierInterface;
-use Cake\Utility\Hash;
+use Cake\Cache\Cache;
+use Cake\Http\Client;
 use Cake\Utility\Security;
 use Exception;
+use Firebase\JWT\CachedKeySet;
 use Firebase\JWT\JWK;
 use Firebase\JWT\JWT;
 use Firebase\JWT\Key;
+use Laminas\Diactoros\RequestFactory;
 use Psr\Http\Message\ServerRequestInterface;
 use RuntimeException;
 use stdClass;
+use Symfony\Component\Cache\Adapter\Psr16Adapter;
```

Comment:
Why do we need this Symfony adapter? Cake's cache engine are already PSR16 compliant.

Reply:
yes, php-jwt opted for PSR-6 based Cache. Psr16Adapter returns a PSR6 Cache.
See https://symfony.com/doc/current/components/cache/psr6_psr16_adapters.html#using-a-psr-16-cache-object-as-a-psr-6-cache

Reply:
Should we write our own adapter instead?

Reply:
The cake cache adapters already implement the PSR16 interfaces. We could implement a 6 to 16 adapter in cake to avoid adding another medium sized dependency.